components/copy-button.tsx

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from "react";
+
+import { ClipboardCopyIcon } from "@heroicons/react/outline";
+
+export function CopyButton({ text }: { text: string }) {
+  const [isCopied, setCopied] = useState<boolean>(false);
+  const [canCopy, setCanCopy] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (
+      typeof window === "object" &&
+      typeof navigator.clipboard === "object" &&
+      typeof navigator.clipboard.writeText === "function"
+    ) {
+      setCanCopy(true);
+    }
+  }, []);
+
+  if (!canCopy) return null;
+
+  function handleCopy(e: React.FormEvent<HTMLButtonElement>) {
+    e.currentTarget.blur();
+    return navigator.clipboard
+      .writeText(text)
+      .then(() => {
+        setCopied(true);
+        setTimeout(() => {
+          setCopied(false);
+        }, 1000);
+      })
+      .catch((err) => console.error(err));
+  }
+
+  return (
+    <button
+      aria-label="salin"
+      className="relative z-10 w-8 h-6 ml-2 bg-gray-100 text-gray-600 hover:bg-blue-600 hover:text-white flex-none flex items-center justify-center rounded"
+      onClick={handleCopy}
+      type="button"
+    >
+      {isCopied ? <span>✓</span> : <ClipboardCopyIcon className="h-4 w-4" />}
```

Comment:
```suggestion
      {isCopied ? <span>Tersalin ✓</span> : <span>Salin <ClipboardCopyIcon className="h-4 w-4" /></span>}
```

I think adding a word here helps in:

1. Making the actions more obvious to the users
2. Increasing the tap area so it's easier to tap using a thumb on mobile devices

Reply:
Sure good idea @zainfathoni , but I switched the icon-text order in the last push. Now lets wait for the deploy... 😅